api-metastore/src/main/java/org/zalando/nakadi/controller/PartitionsController.java

```diff
@@ -103,59 +111,174 @@ private static NakadiCursor selectLast(final List<Timeline> activeTimelines, fin
+    }
+
+    private List<EventTypePartitionView> getViewsByMatchingSamePartitionAcrossStats(
+            final List<Timeline> timelines,
+            final List<PartitionStatistics> filteredFirstStats,
+            final List<PartitionStatistics> lastStats) {
+        //For each partition in first stat, find the matching one in last stat
+        //and create EventTypePartitionView
+        //special handling for finding correct last  partition is done in selectLast()
+        return filteredFirstStats.stream().map(first -> {
             final PartitionStatistics last = lastStats.stream()
                     .filter(l -> l.getPartition().equals(first.getPartition()))
                     .findAny().get();
             return new EventTypePartitionView(
-                    eventTypeName,
                     first.getPartition(),
                     cursorConverter.convert(first.getFirst()).getOffset(),
                     cursorConverter.convert(selectLast(timelines, last, first)).getOffset());
         }).collect(Collectors.toList());
-        return ok().body(result);
+    }
+
+    private List<Cursor> getParsedCursors(final String cursorsString, final String eventType){
+        if(StringUtils.isEmpty(cursorsString)){
+            return emptyList();
+        }
+        try {
+            final List<Cursor> cursorList = objectMapper.
+                    readValue(
+                        cursorsString,
+                        new TypeReference<List<Cursor>>() {
+                    });
+
+            validateCursors(cursorList, eventType);
+            return cursorList;
+        } catch (JsonProcessingException e) {
+            throw new UnparseableCursorException("malformed cursors", e, cursorsString);
+        }
+
+    }
+
+    private void validateCursors(final List<Cursor> cursorList, final String eventType) {
+        for (final Cursor cursor : cursorList) {
+            if (StringUtils.isEmpty(cursor.getPartition())) {
+                throw new InvalidCursorException(CursorError.NULL_PARTITION, cursor, eventType);
+            } else if (StringUtils.isEmpty(cursor.getOffset())) {
+                throw new InvalidCursorException(CursorError.NULL_OFFSET, cursor, eventType);
+            }
+        }
+    }
+
+    private List<PartitionStatistics> initializeLastStats( final List<Timeline> timelines,
+                                                           final List<PartitionStatistics> firstStats) {
+        final List<PartitionStatistics> lastStats;
+        if (timelines.size() == 1) {
+            lastStats = firstStats;
+        } else {
+            lastStats = timelineService.
+                    getTopicRepository(timelines.get(timelines.size() - 1))
+                    .loadTopicStatistics(singletonList(timelines.get(timelines.size() - 1)));
+        }
+        return lastStats;
+    }
+
+    private List<EventTypePartitionView> getViewsWithUnconsumedEvents
+            (final String eventTypeName,
```

Comment:
Why the line break before argument list?

Reply:
I think I was concerned about the 120 char limit for a line, but now i see that keeping the first argument on the same line does not exceed it.